File: 2021-10-14/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int majority_element(vector<int> nums){
	unordered_map<int, int> m;
	for(auto n:nums)
		m[n]++;
	for(auto x:m)
		if(x.second>nums.size()/2)
			return x.first;
	cout << "No majority element found" << endl;
	return -1;
}

signed main() {
	cout << majority_element({3, 5, 3, 3, 2, 4, 3});

	return 0;
}

File: 2021-10-14/solution.py
def majority_element(nums):
	count = {}
	for n in nums:
		if n in count:
			count[n] += 1
		else:
			count[n] = 1
	for key, value in count.items():
		if value>len(nums)//2:
			return key
	return "No majority element found"

print(majority_element([3, 5, 3, 3, 2, 4, 3]))